packages/shared/src/contexts/SettingsContext.tsx

```diff
@@ -174,7 +177,17 @@ export const SettingsContextProvider = ({
   };
 
   const syncSettings = async (bootUserId?: string) => {
-    await updateRemoteSettingsFn(settings, bootUserId);
+    const updatedSettings = settings;
+
+    if (settings.sidebarExpanded) {
+      const sidebarClosed = getFeatureValue(feature.sidebarClosed);
```

Comment:
Here we could have actually used the conditionalFeature but also happy with this. 👍👍

Reply:
My concern was that the user might have not fully registered yet but might get enrolled. So I thought of using this.

Reply:
The timing would be the same though 👀 but this is fine for me 👍

Reply:
I think the reason we pass `bootUserId` on this function is that the user object (from the auth context hook) might not be updated yet which could also be the case if we call the feature flag from the hook. But yeah, they both might work similarly.

Reply:
Oh but then does the useFeatureValue use the right user id?

This might be crucial for the analytics parts so rather double check/inform the analytics team if its some nuance 🙏